web3.go

```diff
@@ -336,3 +375,81 @@ func ParseLogs(myabi abi.ABI, logs []*types.Log) ([]Event, error) {
 	}
 	return output, nil
 }
+
+func ParseAmount(amount string) (*big.Int, error) {
+	var ret = new(big.Int)
+	amount = strings.ToLower(amount)
+	switch {
+	case strings.Contains(amount, "nanogo"):
+		arr := strings.Split(amount, "nanogo")
+		if len(arr) != 2 || arr[0] == "" {
+			return nil, errors.New("Cannot parse the amount, use 'Xnanogo' where X is a number")
+		}
+		val, err := ParseBigInt(arr[0])
+		if err != nil {
+			return nil, err
+		}
+		return ret.Mul(val, weiPerGwei), nil
+	case strings.Contains(amount, "gwei"):
+		arr := strings.Split(amount, "gwei")
+
+		if len(arr) != 2 || arr[0] == "" {
+			return nil, errors.New("Cannot parse the amount, use 'Xgwei' where X is a number")
+		}
+		val, err := ParseBigInt(arr[0])
+		if err != nil {
+			return nil, err
+		}
+		return ret.Mul(val, weiPerGwei), nil
+	case strings.Contains(amount, "attogo"):
+		arr := strings.Split(amount, "attogo")
+		if len(arr) != 2 || arr[0] == "" {
+			return nil, errors.New("Cannot parse the amount, use 'Xattogo' where X is a number")
+		}
+		val, err := ParseBigInt(arr[0])
+		if err != nil {
+			return nil, err
+		}
+		return val, nil
+	case strings.Contains(amount, "wei"):
+		arr := strings.Split(amount, "wei")
+		if len(arr) != 2 || arr[0] == "" {
+			return nil, errors.New("Cannot parse the amount, use 'Xwei' where X is a number")
+		}
+		val, err := ParseBigInt(arr[0])
+		if err != nil {
+			return nil, err
+		}
+		return val, nil
+	default:
+		arr := strings.Split(amount, "go")
+		if len(arr) == 2 && arr[0] != "" {
+			val, err := ParseBigInt(arr[0])
+			if err != nil {
+				return nil, err
+			}
+			return ret.Mul(val, weiPerGO), nil
+		}
+		arr = strings.Split(amount, "eth")
+		if len(arr) == 2 && arr[0] != "" {
+
+			val, err := ParseBigInt(arr[0])
+			if err != nil {
+				return nil, err
+			}
+			return ret.Mul(val, weiPerGO), nil
+		}
+		return nil, errors.New("Cannot parse the amount, use 'Xgo' or 'Xeth' where X is a number")
+	}
+}
+
+func ParseBigInt(value string) (*big.Int, error) {
+	if value == "" {
+		return nil, nil
+	}
```

Comment:
Can we get rid of this blank->nil and return an error instead? It's left over from a very old version of this function when it was doing something much different and more complicated. Otherwise we need nil checks after every call, and if we have to check it anyways then it might as well be an empty string check before calling. This will also make accidental misuse safer since it will return an error (and they must already handle that path) instead of panicking.

Reply:
Should probably document that behavior as well as that it expects base 10, not hex.